src/provider/testSourceProvider.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { RelativePattern, Uri, workspace, WorkspaceFolder } from 'vscode';
+import { getTestSourcePaths } from '../utils/commandUtils';
+
+class TestSourcePathProvider {
+    /**
+     * Test source paths which contains all the source paths from a general project(eclipse & unmanaged-folder)
+     * even it does not have 'test' attribute.
+     */
+    private testSource: string[];
+    /**
+     * Test source paths of which the classpath entry have 'test' attribute.
+     */
+    private strictTestSource: string[];
+
+    public async initialize(): Promise<void> {
+        this.testSource = [];
+        this.strictTestSource = [];
+        if (!workspace.workspaceFolders) {
+            return;
+        }
+
+        this.testSource = await getTestSourcePaths(workspace.workspaceFolders.map((workspaceFolder: WorkspaceFolder) => workspaceFolder.uri.toString()), true);
+        this.strictTestSource = await getTestSourcePaths(workspace.workspaceFolders.map((workspaceFolder: WorkspaceFolder) => workspaceFolder.uri.toString()), false);
```

Comment:
For simplicity it's ok. But actually these are uplicate calls, same check is executed on same workspaceFolders. Consider to refine the implementation if it's expensive, e.g. return both in one call, one check.

Reply:
make sense